pkg/httpclient/http_client.go

```diff
@@ -22,13 +23,23 @@ type httpClient struct {
 
 var slog = logrus.WithField("module", "server")
 
-func NewHTTPClient(timeout time.Duration, overrideHeaders map[string][]string, skipTLS bool) Caller {
+func NewHTTPClient(timeout time.Duration, overrideHeaders map[string][]string, skipTLS bool, capath string) Caller {
 	transport := &http.Transport{
 		DialContext:     (&net.Dialer{Timeout: timeout}).DialContext,
 		IdleConnTimeout: timeout,
 	}
 
-	//TODO: add loki tls config https://issues.redhat.com/browse/NETOBSERV-309
+	if capath != "" {
+		caCert, err := ioutil.ReadFile(capath)
+		if err != nil {
+			slog.Errorf("Cannot load loki ca certificate: %v", err)
+		} else {
+			pool := x509.NewCertPool()
+			pool.AppendCertsFromPEM(caCert)
+			transport.TLSClientConfig = &tls.Config{RootCAs: pool}
+		}
+	}
+
 	if skipTLS {
```

Comment:
`skipTLS` may fail if `capath` is set and `ReadFile` fails. We should split both like:

```go
if skipTLS {
  ...
} else if capath != "" {
  ...
}
```

Reply:
I would say if the path is invalid / file missing. And yes it's easier to read :+1:

Reply:
No, I was saying why skipTLS would have failed on ReadFile error?

Reply:
Thought there was a return after the `Errorf` but seems not. My bad :angel: 

Thanks !